Functions/QualityControl/xASL_qc_GenerateReport.m

```diff
@@ -0,0 +1,418 @@
+        if strcmp(PatientInfo{1, iEntry},'participant_id')
+            line = xASL_sub_PrintText(['Participant: ', PatientInfo{2, iEntry}], figure, line, settings );
+        elseif strcmp(PatientInfo{1, iEntry},'age')
+            line = xASL_sub_PrintText(['Participant age: ', num2str(PatientInfo{2, iEntry})], figure, line, settings );
+        elseif strcmp(PatientInfo{1, iEntry},'sex')
+            line = xASL_sub_PrintText(['Participant sex: ', PatientInfo{2, iEntry}], figure, line, settings );
+        end
+    end
+
+end
+
+function line = xASL_sub_PrintQC(json, x, figure, line, settings)
+    % Stop if module and field don't exists in output
+    if ~isfield(x.Output, (json.module)) && ~isfield(x.Output.(json.module), json.parameter)
+        return     
+    end
+
+    % Use field specific settings if they exists
+    settings.fontName = 'monospace';
+    if isfield(json, 'settings')
+        settings = xASL_sub_loadSettings(json.settings, settings);
+    end
+
+    TempValue = x.Output.(json.module).(json.parameter);
+    
+    if ~isfield(json, 'alias') 
+        json.alias = json.parameter;
+    end
+
+    if ~isfield(json, 'unit') 
+        json.unit = '';
+    end
+
+    if isfield(json, 'range') 
+        [range] = strsplit(json.range, '-');
+        if (TempValue < xASL_str2num(range{1})) || (TempValue > xASL_str2num(range{2}))
+            settings.color = 'r';
+        end
+        json.range = ['(' json.range ')'];
+    else
+        json.range = '';
+    end
+
+    if isnumeric(TempValue)
+        TempValue = num2str(TempValue);
+    end
+
+    if size(TempValue, 1) == 1
+        TextString = sprintf([sprintf('%-20s',[json.alias,':']), sprintf('%8s',TempValue),sprintf('%-12s', [json.unit, ' ' , json.range]), ' \n']);
+    end
+
+    line = xASL_sub_PrintText(TextString, figure, line, settings);
+
+end
+
+function [settings] = xASL_sub_PrintImage(input, x, figure, settings, position)
+    switch nargin
+    case 4
+        position = [xASL_str2num(input.position) xASL_str2num(input.size)];
+
+        if isfield(input, 'xPath') && isfield(x.P, input.name)
+            ImagePath = x.P.(input.name);
+        elseif isfield(input, 'absolutePath')
+            ImagePath = input.absolutePath;
+        elseif isfield(input, 'popPath')
+            ImagePath = fullfile(x.dir.xASLDerivatives, 'Population', input.popPath);
+            ImagePath = xASL_sub_replace(ImagePath, x);
+        elseif isfield(input, 'subjPath')
+            ImagePath = fullfile(x.dir.xASLDerivatives, x.SUBJECT, input.subjPath);
+        else
+            warning('xASL_sub_PrintImage didnt have a defined path') ;
+            return
+        end
+
+        if isfield(input, 'header')
+            header = input.header;
+        else
+            header = '';
+        end
+    case 5
+        ImagePath = input;
+        header = '';
+    end
+
+    [canvas, ~] = xASL_sub_createNewCanvas(position(1:2), position(3:4), settings.canvas);
```

Comment:
~ can be removed. is only necessary when wanting to take the second or other output but ignoring the first